app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -3,8 +3,51 @@ kind: registration
 name: Global Registration
 model: ProvisioningTemplate
 -%>
+<%
+    curl_opts = ["-H 'Content-Type: application/json'", "-H 'Accept: application/json'", "-H 'Authorization: Bearer #{@auth_token}'"]
+%>
+#!/bin/bash
+
+# Template parameters:
+<%= "#  - User: [#{@user.login}]" if @user %>
+<%= "#  - Organization: [#{@organization.name}]" if @organization %>
+<%= "#  - Location: [#{@location.name}]" if @location %>
+<%= "#  - Hostgroup: [#{@hostgroup.name}]" if @hostgroup %>
 
 if ! [ $(id -u) = 0 ]; then
     echo "Please run as root"
     exit 1
 fi
+
+if [ -f /etc/os-release ] ; then
+  . /etc/os-release
+fi
+
+HOSTNAME=$(hostname -f)
+
+echo "#"
+echo "# Running registration"
+echo "# Hostname: $HOSTNAME"
+echo "# OS: $NAME"
+echo "#"
+
+if [ $ID = rhel ] ; then
+    echo "RHEL registration"
+<% if plugin_present?('katello') %>
+    CONSUMER_RPM=$(mktemp).rpm
```

Comment:
it's safer to use a template than to concat. That way mktemp can check if the file really doesn't exist, rather than part of the filename.
```suggestion
    CONSUMER_RPM=$(mktemp tmp.XXXXXXXXXX.rpm)
```

Reply:
Almost sure it is GNU

Reply:
Me too. At least https://www.tutorialspoint.com/unix_commands/mktemp.htm doesn't mention it. However, the template is AFAIK universal so that's what I'd recommend.